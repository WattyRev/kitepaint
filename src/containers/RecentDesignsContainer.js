import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import designShape from "../models/design";
import { getRecentDesigns } from "../redux/modules/designs";
import { GET_DESIGNS } from "../redux/actions";

/**
 * Provides information and actions about/for Designs.
 */
export class RecentDesignsContainer extends React.Component {
  static propTypes = {
    children: PropTypes.func.isRequired,
    designs: PropTypes.arrayOf(designShape).isRequired,
    getDesigns: PropTypes.func.isRequired
  };

  componentDidMount() {
    // Get the designs, but limit the request to 6 since we don't need to show many.
    this.props.getDesigns({
      limit: 6
    });
  }

  render() {
    return this.props.children({
      actions: {},
      props: {
        designs: this.props.designs
      }
    });
  }
}

const mapStateToProps = state => ({
  designs: getRecentDesigns(state)
});

const mapDispatchToProps = {
  getDesigns: GET_DESIGNS
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(RecentDesignsContainer);
