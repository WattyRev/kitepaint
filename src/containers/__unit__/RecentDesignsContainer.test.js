import React from "react";
import { shallow } from "enzyme";
import { RecentDesignsContainer } from "../RecentDesignsContainer";

describe("RecentDesignsContainer", () => {
  let props;
  beforeEach(() => {
    props = {
      designs: [],
      getDesigns: jest.fn()
    };
  });
  it("renders", () => {
    expect.assertions(1);
    const wrapper = shallow(
      <RecentDesignsContainer {...props}>
        {() => <div>test</div>}
      </RecentDesignsContainer>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("triggers getDesigns when mounting", () => {
    expect.assertions(1);
    shallow(
      <RecentDesignsContainer {...props}>
        {() => <div>test</div>}
      </RecentDesignsContainer>
    );
    expect(props.getDesigns.mock.calls).toHaveLength(1);
  });
  it("provides the designs in the render", () => {
    expect.assertions(1);
    props.designs = [
      {
        id: "123",
        created: "10/01/01",
        name: "my-design",
        variations: []
      }
    ];
    const wrapper = shallow(
      <RecentDesignsContainer {...props}>
        {data => (
          <div className="testing_target">
            {data.props.designs.map(design => design.name)}
          </div>
        )}
      </RecentDesignsContainer>
    );
    expect(wrapper.find(".testing_target").text()).toEqual("my-design");
  });
});
