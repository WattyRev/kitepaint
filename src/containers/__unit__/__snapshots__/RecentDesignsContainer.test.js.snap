// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentDesignsContainer renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecentDesignsContainer
    designs={Array []}
    getDesigns={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "limit": 6,
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  >
    [Function]
  </RecentDesignsContainer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "test",
    },
    "ref": null,
    "rendered": "test",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
      },
      "ref": null,
      "rendered": "test",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
